importer/src/main/java/org/hiero/mirror/importer/parser/record/historicalbalance/HistoricalBalanceService.java

```diff
@@ -206,4 +219,25 @@ private boolean shouldGenerate(long consensusEnd) {
                                 >= properties.getMinFrequency().toNanos())
                         .isPresent();
     }
+
+    private void checkTreasuryAccount() {
+        var treasuryAccountEntityId = systemEntity.treasuryAccount();
+        var treasuryAccount = entityRepository.findById(treasuryAccountEntityId.getId());
+        if (treasuryAccount.isEmpty()) {
+            log.warn("Treasury account {} not found in entity table, creating it", treasuryAccountEntityId);
+            entityRepository.save(Entity.builder()
+                    .balance(0L)
+                    .createdTimestamp(0L)
+                    .declineReward(true)
+                    .deleted(false)
+                    .id(treasuryAccountEntityId.getId())
+                    .memo("SENTINEL_AUTO_GENERATED")
```

Comment:
Would prefer a bit more explanatory memo.

```suggestion
                    .memo("Mirror node created synthetic treasury account")
```